vavr/src/main/java/io/vavr/collection/Stream.java

```diff
@@ -234,6 +234,20 @@
         return Stream.ofAll(Iterator.iterate(seed, f));
     }
 
+    /**
+     * Generates a (theoretically) infinitely long Stream using a repeatedly invoked supplier
+     * that provides a {@code Some} for each next value and a {@code None} for the end
+     *
+     * @param supplier A Supplier of iterator values
+     * @param <T> value type
+     * @param <A> supplier option type
+     * @return A new Stream
+     */
+    static <T, A extends T> Stream<T> iterate(Supplier<Option<A>> supplier) {
+        Objects.requireNonNull(supplier, "supplier is null");
+        return Stream.ofAll(Iterator.iterate(supplier));
```

Comment:
Since `Stream` (contrary to `Iterator`) can be iterated multiple times, I think it's worth mentioning in the javadoc that the Stream will constructed in such a way that the `Supplier` will be invoked only that many times until it returns `None`, and repeated iteration over the stream will produce the same values in the same order, without any further invocations to the `Supplier`.

Reply:
thanks, done